dashboard/app/models/reference_guide.rb

```diff
@@ -18,4 +18,70 @@
 #  index_reference_guides_on_course_version_id_and_parent_key  (course_version_id,parent_reference_guide_key)
 #
 class ReferenceGuide < ApplicationRecord
+  belongs_to :course_version
+  validates_uniqueness_of :key, scope: :course_version_id
```

Comment:
I think it would be good to add some validation around what characters can be in the key. For most keys we allow letters, numbers and dashes only.